src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -311,6 +313,243 @@ public void Dispose()
             this.Dispose(disposing: true);
             GC.SuppressFinalize(this);
         }
+
+        #region "Asynchronous API"
+
+        public async Task StartPaddingFunctionScopeAsync()
+        {
+            await this.FlushAsync().ConfigureAwait(false);
+            await this.writeStream.WriteAsync(parentheses[..1]).ConfigureAwait(false);
+        }
+
+        public async Task WritePaddingFunctionNameAsync(string functionName)
+        {
+            await this.FlushAsync().ConfigureAwait(false);
+            await this.writeStream.WriteAsync(Encoding.UTF8.GetBytes(functionName)).ConfigureAwait(false);
+        }
+
+        public async Task EndPaddingFunctionScopeAsync()
+        {
+            await this.FlushAsync().ConfigureAwait(false);
+            await this.writeStream.WriteAsync(parentheses[1..2]).ConfigureAwait(false);
+        }
+
+        public async Task StartObjectScopeAsync()
+        {
+            this.writer.WriteStartObject();
+            await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+        }
+
+        public async Task EndObjectScopeAsync()
+        {
+            this.writer.WriteEndObject();
+            await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+        }
+
+        public async Task StartArrayScopeAsync()
+        {
+            this.writer.WriteStartArray();
+            await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+        }
+
+        public async Task EndArrayScopeAsync()
+        {
+            this.writer.WriteEndArray();
+            await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+        }
+
+        public async Task WriteNameAsync(string name)
+        {
+            this.writer.WritePropertyName(name);
+            await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+        }
+
+        public async Task WriteValueAsync(bool value)
+        {
+            this.writer.WriteBooleanValue(value);
+            await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+        }
+
+        public async Task WriteValueAsync(int value)

```

Comment:
so these methods are just wrappers of the synchronous methods?

Reply:
They wrap the synchronous methods of the inner `Utf8JsonWriter` instance. The `Utf8JsonWriter` exposes only synchronous Write methods. This is because it exclusively writes to memory, it does not write directly to stream, so it never performs any I/O operations via its Write methods that need to be async. That's why after each `Write***Value` I call the `FlushIfBufferThresholdReachedAsync()` method which will write the buffered data to the stream if the buffer has reached a certain threshold. Because this method potentially writes to a stream, it is async.